test/can_message_tests.cpp

```diff
@@ -0,0 +1,71 @@
+//
+// Created by abyl on 14.02.24.
+//
+
+#include <gtest/gtest.h>
+#include "isobus/hardware_integration/can_hardware_interface.hpp"
+#include "isobus/hardware_integration/virtual_can_plugin.hpp"
+#include "isobus/isobus/can_message.hpp"
+#include "isobus/isobus/can_message_frame.hpp"
+#include "isobus/isobus/can_network_manager.hpp"
+
+using namespace isobus;
+
+std::uint64_t value64;
+std::uint16_t value16;
+
+void callback(const CANMessage &message, void *parent)
+{
+	value16 = message.get_int16_at(0);
+	EXPECT_EQ(value16, 513);
+	value16 = message.get_int16_at(0, CANMessage::ByteFormat::BigEndian);
+	EXPECT_EQ(value16, 258);
+	value64 = message.get_int64_at(0);
+	EXPECT_EQ(value64, 578437695752307201);
+	value64 = message.get_int64_at(0, CANMessage::ByteFormat::BigEndian);
+	EXPECT_EQ(value64, 72623859790382856);
+	value64 = message.get_data_custom_length(8, 16);
+	EXPECT_EQ(value64, 770);
+	value64 = message.get_data_custom_length(8, 16, CANMessage::ByteFormat::BigEndian);
+	EXPECT_EQ(value64, 515);
+	value64 = message.get_data_custom_length(8, 15);
+	EXPECT_EQ(value64, 258);
+	value64 = message.get_data_custom_length(8, 15, CANMessage::ByteFormat::BigEndian);
+	EXPECT_EQ(value64, 513);
+	value64 = message.get_data_custom_length(14, 3);
+	EXPECT_EQ(value64, 4);
+	value64 = message.get_data_custom_length(14, 3, CANMessage::ByteFormat::BigEndian);
+	EXPECT_EQ(value64, 4);
+	EXPECT_THROW({ value64 = message.get_data_custom_length(63, 999999); }, std::out_of_range);
+	EXPECT_THROW({ value64 = message.get_data_custom_length(65748321, 1); }, std::out_of_range);
```

Comment:
I think generally I'd rather add boundary checks rather than relying on an exception. The library broadly does not use exceptions, and prefers to soft-fail on bad information rather than crash the program. Doing it the way you have it here 
 makes a consumer of the library have to know the implementation of this function to avoid causing a crash which seems not-deal.

Reply:
Okay, fixed it. I didn't have a lot of time working on this but slowly managing to go through at times.